include/bitcoin/bitcoin/utility/serializer.hpp

```diff
@@ -83,6 +83,9 @@ class serializer
     /// Write all bytes.
     void write_bytes(const data_chunk& data);
 
+    /// Write all bytes.
+    void write_bytes(const data_slice data);
```

Comment:
As you prefer. I thought maybe in some low level db code, that the pointer misdirection might result in cache misses but maybe that premature optimization.

Reply:
There would be no issue with data consistency, I think I just avoided `data_slice` here for store perf reasons. But I avoid having redundant `data_chunk` and `data_slice` overloads.

Reply:
Note that this also affects the `writer` interface.

Reply:
And therefore also the `ostream_writer` (and stream tests).

Reply:
I don't think there is any perf difference at all however.

Reply:
ok changing to data_slice

Reply:
Remember to remove const from the data slice parameter. For reasons described elsewhere this is avoided because the parameter is always constructed (ie passed by copy) and as such the type is not made safer by const, making it superfluous (unnecessary verbosity).

FWIW we always use const where possible in variable (vs. parameter) declarations because it declares that the intent is the variable not change in the scope, which can help avoid subtle mistakes (ie provides some type safety vale), but as a local variable is easily changed as necessary. Overconstraining a parameter (ie with const) has the opposite effect as it does not provide any interface safety but constrains internal implementation (as we should not modify interfaces once they are published). The same logic is applied to members, always const where possible. The one annoyance is that const members don’t work when implementing copy assignment (even though conceptually you are constructing a new instance).

Reply:
Thanks for the explanation, I can see that all the data members and variables are declared const.  Still the const is maybe desirable for readability and enforcement of constness if there's future changes. Readability shows that this is an object whose data is not meant to change. I find that a really useful indicator in code.

Reply:
It’s not incorrect to use it, but it is superfluous apart from providing the hint that the value won’t be changed. However if the caller requires a variable not be changed a const variable can be passed. This stylistic preference is consistently applied to parameters passed by copy (including all data_slice) so that consistency should be maintained here if only as a matter of consistency.